File: CleanAddisBackOfficeApi/src/main/java/com/iyoa/cleanaddis/repository/ArticleRepository.java
package com.iyoa.cleanaddis.repository;

import java.util.Date;
import java.util.List;
import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import com.iyoa.cleanaddis.model.Article;


public interface ArticleRepository extends JpaRepository<Article, UUID> {


	
	
}

File: app/src/main/java/com/iyoa/cleanaddis/repository/news/ArticleRepository.kt
package com.iyoa.cleanaddis.repository.news

import androidx.lifecycle.LiveData
import com.iyoa.cleanaddis.data.news.Article
import com.iyoa.cleanaddis.data.news.ArticleDAO

class ArticleRepository(private val articleDAO: ArticleDAO) {
    fun allArticles(): LiveData<List<Article>> = articleDAO.getAllNews()

    fun insertArticles(article: Article){
        articleDAO.insertArticle(article)
    }

    fun addArticles(listOfarticles: List<Article>) {
        articleDAO.addArticles(listOfarticles)
    }

    fun getArticle(uuid:String):LiveData<Article>{

        return articleDAO.getNewsByUuid(uuid)
    }
}